march_hardware/src/IMotionCube.cpp

```diff
@@ -331,6 +361,33 @@ void IMotionCube::reset()
   sdo_bit16_write(this->slaveIndex, 0x2080, 0, 1);
 }
 
+uint16_t IMotionCube::computeSWCheckSum(int& start_address, int& end_address)
+{
+  size_t pos = 0;
+  size_t old_pos = 0;
+  uint16_t sum = 0;
+  std::string delimiter = "\n";
+  std::string token;
+  while ((pos = sw_stream_.str().find(delimiter, old_pos)) != std::string::npos)
```

Comment:
There is no use in using a `std::stringstream` if the first thing you are going to do is reading out the entire stream. Streams are there for data that is not yet complete, for example, reading from files. Therefore, if you need the complete string in order to compute the checksum, you shouldn't be using streams, but strings. I think the implementation would become much lighter if you use a `std::string`